projects/utils.py

```diff
@@ -28,3 +31,29 @@ def filter_projects(queryset, data, request):
         queryset = filter_by_budget(queryset, budget_filter)
 
     return queryset
+
+
+def calculate_difference(project1, project2):
+    difference = {}
+    for field in project1._meta.fields:
+        field_name = field.name
+        if getattr(project1, field_name) != getattr(project2, field_name):
+            if field_name == 'industry':
+                industry_difference = compare_industries(project1.industry, project2.industry)
+                difference[field_name] = industry_difference
+            else:
+                difference[field_name] = {
+                    'project1': getattr(project1, field_name),
+                    'project2': getattr(project2, field_name)
+                }
+    return difference
+
+
```

Comment:
please put the function compare_industries(industry1, industry2) before the function calculate_difference(project1, project2)please put the function compare_industries(industry1, industry2) before the function calculate_difference(project1, project2)

Reply:
done